symengine/type_codes.h

```diff
@@ -0,0 +1,57 @@
+SYMENGINE_ENUM(INTEGER),
```

Comment:
We should see how other projects do this, but strictly speaking this file is not a header file anymore, since it won't work if you just include it. So perhaps we should rename it to `type_codes.inc`, or something similar.